web/pages/api/signup.js

```diff
@@ -0,0 +1,33 @@
+import nextConnect from 'next-connect'
+import bcrypt from 'bcrypt'
+import { query } from '../../utils/query'
+
+const handler = nextConnect()
+
+    .post(async (req, res) => {
+        const { email, password } = await req.body 
+        
+        if (!email || !password) res.json({ error: 'Email or Password cannot be blank!' })
+
+        try {
+
+          /*
+            todo:
+              - validate password: see if it has 8 digits and 1 symbol 
+              - add check to see if email has @uwinnipeg.ca or @webmail.uwinnipeg.ca
+              - add check to see if email exists
+              - send validation link to email
+          */
+
+            const hashedPW = await bcrypt.hash(password, 10)
```

Comment:
The salt rounds should probably be a system constant somewhere, in case we decide to change it later for some reason.

Reply:
added now